src/django_upgrade/fixers/on_delete.py

```diff
@@ -6,61 +6,125 @@
 
 import ast
 from functools import partial
-from typing import Iterable
+from typing import Iterable, MutableMapping
+from weakref import WeakKeyDictionary
 
 from tokenize_rt import Offset, Token
 
-from django_upgrade.ast import ast_start_offset
+from django_upgrade.ast import ast_start_offset, is_rewritable_import_from
 from django_upgrade.data import Fixer, State, TokenFunc
-from django_upgrade.tokens import OP, find, insert, parse_call_args
+from django_upgrade.tokens import (
+    OP,
+    erase_node,
+    extract_indent,
+    find,
+    insert,
+    parse_call_args,
+)
 
 fixer = Fixer(
     __name__,
     min_version=(1, 9),
 )
 
 
+@fixer.register(ast.ImportFrom)
+def visit_ImportFrom(
+    state: State,
+    node: ast.ImportFrom,
+    parents: list[ast.AST],
+) -> Iterable[tuple[Offset, TokenFunc]]:
+    if (
+        node.module == "django.db.models"
+        and is_rewritable_import_from(node)
+        and any(alias.name in {"ForeignKey", "OneToOneField"} for alias in node.names)
+    ):
+        yield ast_start_offset(node), partial(
+            update_django_models_import,
+            node=node,
+            state=state,
+        )
+
+
+# Track if we need to update `from django.db.models` import to add CASCADE.
+should_update_import: MutableMapping[State, bool] = WeakKeyDictionary()
+
+
+def update_django_models_import(
+    tokens: list[Token], i: int, *, node: ast.ImportFrom, state: State
+) -> None:
+    if should_update_import.pop(state, False):
+        used_names = state.from_imports["django.db.models"]
+        used_names.add("CASCADE")
+
+        j, indent = extract_indent(tokens, i)
+        erase_node(tokens, i, node=node)
+        joined_names = ", ".join(sorted(used_names))
+        insert(
+            tokens,
+            j,
+            new_src=f"{indent}from django.db.models import {joined_names}\n",
+        )
```

Comment:
A noble idea to try and nicely sort the imports, but unfortunately this will erase comments, and will undo the "one import per line" style

Generally users should be able to rely on their imoprt sorter (isort, reorder_python_imports) to fix up any non-compliant import style we add. So I'd rather do the most conservative definitely-will-work thing.

I am changing this to instead just insert `from django.db.models import CASCADE` on a line above.

We could also look at refactoring `update_imports` to add extra imports at the start, but that could be a separate issue.